modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Checkbox/Checkbox.es.js

```diff
@@ -19,6 +19,23 @@ import React, {useEffect, useState} from 'react';
 import {FieldBase} from '../FieldBase/ReactFieldBase.es';
 import {useSyncValue} from '../hooks/useSyncValue.es';
 
+const toBoolean = (value) => {
+	const valueType = typeof value;
+	let booleanValue, newValue;
+
+	switch (valueType) {
+		case 'object':
+			[newValue] = value;
+			booleanValue = newValue === 'true';
+			break;
+
+		default:
+			return value;
+	}
+
+	return booleanValue;
+};
```

Comment:
I not sure if I got the idea here... can you explain what you it is doing?

Reply:
Basically, the implementation of `Boolean Field` of the `Forms Sidebar` changes the component of `predefined value` of a `Switch` to a Select, therefor this function is responsible for convert the array returned by `Select` in a boolean value.

Reply:
I think that this could be best.

Reply:
I talked with @miltonmc on the Slack.